tsl/src/continuous_aggs/Readme.md

```diff
@@ -1,144 +1,115 @@
-
-5. invalidation log: RowExclusive
-6. completed threshold: ??? (as this is only touched by the materialization
-   worker, and those are excluded by the lock on the materialization table, it
-   is unclear if it matters)
-7. invalidation threshold: AccessShareLock
-
-
-The INSERT/UPDATE/DELETE path grabs:
-
-1. raw hypertable: RowExclusive or stronger
-2. Invalidation Threshold: AccessShare (optional)
-3. invalidation log: RowExclusive (note: this ordering does not conflict with
-   the materialization worker's second transaction, because both transactions
-   only grab a row-exclusive lock on the invalidation log, so the two locks
-   don't conflict)
-
-SELECTs grab
-
-1. the user view: AccessShare?
-2. the raw hypertable: the same
+A continuous aggregate is a special kind of materialized view for
+aggregates that can be partially and continuously refreshed, either
+manually or automated by a policy that runs in the background. Unlike
+a regular materialized view, a continuous aggregate doesn't require
+complete re-materialization on every refresh. Instead, it is possible
+to refresh a subset of the continuous aggregate at relatively low
+cost, thus enabling continuous aggregation as new data is written or
+old data is updated and/or backfilled.
+
+To enable continuous aggregation, a continuous aggregate stores
+partial aggregations for every time bucket in an internal
+hypertable. The advantage of this configuration is that each time
+bucket can be recomputed individually, without requiring updates to
+other buckets, and buckets can be combined to form more granular
+aggregates (e.g., hourly buckets can be combined to form daily
+buckets). Finalization of the partial buckets happens automatically at
+query time. Although such finalization gives slightly higher querying
+times, it is offset by more efficient refreshes that only recompute
+the buckets that have been "invalidated" by changes in the raw data.
+
+A continuous aggregate policy automates the refreshing, allowing the
+aggregate to stay up-to-date without manual intervention. A policy can
+be configured to only refresh the most recent data (e.g., just the
+last hour's worth of data) or ensure that the continuous aggregate is
+always up-to-date with the underlying source data. Policies that focus
+on recent data allow older parts of the continuous aggregate to stay
+the same or be governed by manual refreshes.
+
+## Bookkeeping and Internal State ##
+
+TimescaleDB does bookkeeping for each continuous aggregate to know
+which buckets of the aggregates require refreshing. Whenever a
+modification happens to the source data, an invalidation for the
+modified region is written to an invalidation log. However,
+invalidations are not written after the *invalidation threshold*,
+which tracks the latest bucket materialized thus far. This threshold
+allows write amplification to be kept to a minimum by, e.g, never
+refreshing the most recent time bucket that is still being "filled".
```

Comment:
Are there any other things we currently do? Otherwise, just write what we do.
```suggestion
allows write amplification to be kept to a minimum by not
refreshing the most recent time bucket, which is assumed to still have data being added to it.
```